evm/forge/tests/WormholeCctpMessages.t.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.22;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import {IWormhole} from "src/interfaces/IWormhole.sol";
+
+import {WormholeCctpMessages} from "src/libraries/WormholeCctpMessages.sol";
+
+contract MessagesTest is Test {
+    using WormholeCctpMessages for *;
+
+    function test_DepositWithPayloadSerde(
+        bytes32 token,
+        uint256 amount,
+        uint32 sourceCctpDomain,
+        uint32 targetCctpDomain,
+        uint64 cctpNonce,
+        bytes32 burnSource,
+        bytes32 mintRecipient,
+        bytes memory payload
+    ) public {
+        vm.assume(targetCctpDomain != sourceCctpDomain);
+        vm.assume(payload.length > 0);
+        vm.assume(payload.length < type(uint16).max);
+
+        IWormhole.VM memory fakeVaa;
+        fakeVaa.payload = token.encodeDeposit(
+            amount,
+            sourceCctpDomain,
+            targetCctpDomain,
+            cctpNonce,
+            burnSource,
+            mintRecipient,
+            payload
+        );
+        assertEq(fakeVaa.payload.length, 147 + payload.length);
```

Comment:
We should add a comment describing what the 147 byte offset is here.

Reply:
Added a comment in this commit: https://github.com/wormhole-foundation/wormhole-circle-integration/pull/48/commits/3fdb1014a884922660f94a0e545a2c9ec846b870.